Mathlib/Data/Finsupp/Pointwise.lean

```diff
@@ -92,10 +92,11 @@ instance [NonUnitalCommRing β] : NonUnitalCommRing (α →₀ β) :=
   DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
     (fun _ _ ↦ rfl) fun _ _ ↦ rfl
 
--- TODO can this be generalized in the direction of `Pi.smul'`
--- (i.e. dependent functions and finsupps)
--- TODO in theory this could be generalised, we only really need `smul_zero` for the definition
-instance pointwiseScalar [Semiring β] : SMul (α → β) (α →₀ β) where
+-- TODO(Paul-Lez): add a `DFinsupp` version of this.
+-- Note: this creates an instance diamond with `SMul (α → β) (α →₀ (α → β))`, so this is an
+-- abbrev rather than an instance.
+/-- The pointwise scalar multiplication of a function `α → β` on a finsupp `α →₀ β`. -/
+abbrev pointwiseScalar {M : Type*} [Zero M] [SMulZeroClass β M] : SMul (α → β) (α →₀ M) where
```

Comment:
```suggestion
-- def rather than an instance.
/-- The pointwise scalar multiplication of a function `α → β` on a finsupp `α →₀ β`. -/
def pointwiseScalar {M : Type*} [Zero M] [SMulZeroClass β M] : SMul (α → β) (α →₀ M) where
```
Is there any point in making it an `abbrev` rather than `def`?

Reply:
@Paul-Lez I don't think this fits the `[reducible non-instances]` pattern since we aren't constructing the `SMul` instance via some function-constructor, but rather a direct construction. Can you help clarify why this is an `abbrev`?